api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -117,11 +117,22 @@ type NodeGroup struct {
 	// MachineConfigPoolSelector defines label selector for the machine config pool
 	// +optional
 	MachineConfigPoolSelector *metav1.LabelSelector `json:"machineConfigPoolSelector,omitempty"`
+	// NodeSelector defines label selector for the nodes that will host RTE pods
+	// +optional
+	NodeSelector *NodeSelector `json:"nodeSelector,omitempty"`
 	// Config defines the RTE behavior for this NodeGroup
 	// +optional
 	Config *NodeGroupConfig `json:"config,omitempty"`
 }
 
+type NodeSelector struct {
+	// Name the name of the machine config pool
+	Name string `json:"name"`
```

Comment:
is not obvious to me why do we need a name here and most notably the "machine config pool" reference

Reply:
the machine config pool is a copy-paste mistake. 
the name however is needed because we need to identify the ds for such node group. for MCPs we use the value of the nodeslector label of the corresponding MCP. I'm open for suggestions though as stated earlier

Reply:
I think you are right, but the Name should identify a NodeGroup regardless about how we match the nodes (using NodeSelector or MCP), this is why I think it belongs a level above.